tokio-timer/src/interval.rs

```diff
@@ -29,6 +31,19 @@ impl Interval {
         Interval::new_with_delay(Delay::new(at), duration)
     }
 
+    /// Creates new `Interval` that yields with interval of `duration`.
+    ///
+    /// The function is shortcut for `Interval::new(Instant::now() + duration, duration)`.
+    ///
+    /// The `duration` argument must be a non-zero duration.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if `duration` is zero.
+    pub fn interval(duration: Duration) -> Interval {
```

Comment:
Unfortunately, this fn really should be `new`, but we can't just break the existing `new`. Maybe we can name this `new_interval` instead? Not super elegant...

Reply:
Yes I agree with you that original naming is inconvenient.
We could do `new_interval` yeah